pkg/processor/trigger/trigger.go

```diff
@@ -407,6 +395,30 @@ func (at *AbstractTrigger) SignalWorkersToTerminate() error {
 	return nil
 }
 
+func (at *AbstractTrigger) IsBusy() bool {
+	workers := at.Trigger.GetWorkers()
+
+	// should not happen, but just in case
+	if len(workers) == 0 {
+		return false
+	}
+
+	// enough to check only one
+	if !workers[0].IsAsync() {
+		// if the worker is not async, we can check if the number of workers is equal to the number of available objects
+		return len(workers) == at.WorkerAllocator.GetNumObjectsAvailable()
```

Comment:
same here, shouldn't you check if `GetNumObjectsAvailable` is zero?

Reply:
@TomerShor it should be `len(workers) != at.WorkerAllocator.GetNumObjectsAvailable()`